e3sm_diags/driver/utils/dataset_xr.py

```diff
@@ -1249,7 +1277,9 @@ def _get_time_slice(self, ds: xr.Dataset, filename: str) -> slice:
 
         if self.is_sub_monthly:
             start_time = f"{start_yr_str}-01-01"
-            end_time = f"{str(int(end_yr_str) + 1)}-01-01"
+
+            end_yr_str = str(int(end_yr_str) + 1).zfill(4)
+            end_time = f"{end_yr_str}-01-01"
```

Comment:
Fixed a bug where `end_yr_str` would be in the improper format for Xarray subsetting (e.g., `2-01-01`). `zfill(4)` pads the string to ensure it has 4 digits for the year (e.g., `0002-01-01`).